.github/workflows/code-health-tools.yml

```diff
@@ -25,6 +25,31 @@ jobs:
       run: |
         pushd tools/cli
         make build
+  unit-tests:
+    needs: build
+    env:
+      COVERAGE: coverage.out
+      TEST_CMD: gotestsum --junitfile unit-tests.xml --format standard-verbose --
+      UNIT_TAGS: unit
+      INTEGRATION_TAGS: integration
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - name: Install Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+        with:
+          go-version-file: 'tools/cli/go.mod'
+      - name: Install gotest
+        run: go install gotest.tools/gotestsum@ec99a250836f069a524bb9d9b5de0a7a96334ea7
+      - name: Run unit tests
+        run: |
+          pushd tools/cli
```

Comment:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrunworking-directory